FueledUtils/UIExtensions.swift

```diff
@@ -261,6 +261,22 @@ extension UIView {
 		}
 	}
 
+	///
+	/// Renders the view's layer and returns a `UIImage`.
+	///
+	/// - Parameters:
+	///   - opaque: Please refer to the parameters documentation for `UIGraphicsBeginImageContextWithOptions` for more info.
+	///   - scale: Please refer to the parameters documentation for `UIGraphicsBeginImageContextWithOptions` for more info.
+	/// - Returns: The image if it could be generated. If it couldn't, for example if the `UIView`'s width or height is 0, a crash will happen at runtime.
+	///
+	/// - Note: The returns is an implicitely unwrapped optional for backward-compatibility purpose, and will be made an optional in a future release (as well as not crash)
+	///
+	public func renderImage(opaque: Bool = false, scale: CGFloat = 0.0) -> UIImage {
+		return UIImage.draw(size: self.bounds.size, opaque: opaque, scale: scale) { context in
+			self.layer.render(in: context)
```

Comment:
@Sissing Do you remember if this is the best way to do view rendering?

Reply:
@heymansmile does this method render sublayers of the view?

Reply:
@leontiy yes sure it does everything.

Reply:
I think the best way for `UIView` is still `self.drawHierarchy(in:, afterScreenUpdates:)`, as optionally you can just use the current render which can be much faster than re-rendering.
I think this method could make sense in a `CALayer` extension though.

Reply:
@stephanecopin @leontiy this method can't be replaced with the existing one. The existing method works only for visible hierarchy and doesn't work when you, for example, create a custom control and need to apply an effect or gradient to a set of views and then make an image of their superposition in its initializer when it haven't been displayed yet.

Reply:
@heymansmile Can you make it clear in the first line of the descriptions of both methods that one is for views in a presented view hierarchy and the other is for any view, regardless of the hierarchy status.

Reply:
I also think `renderToImage` or `render(...) -> UIImage` would be clearer. The name of the function sounds like it accepts an image as a parameter when it doesn't.